src/renderer/views/pool/PoolDetailsView.tsx

```diff
@@ -72,38 +77,61 @@ export const PoolDetailsView: React.FC = () => {
     }
   }, [oRouteAsset, setSelectedPoolAsset])
 
-  const routeAssetRD = RD.fromOption(oRouteAsset, () =>
-    Error(intl.formatMessage({ id: 'routes.invalid.asset' }, { asset }))
+  const routeAssetRD = useMemo(
+    () => RD.fromOption(oRouteAsset, () => Error(intl.formatMessage({ id: 'routes.invalid.asset' }, { asset }))),
+    [oRouteAsset, asset, intl]
   )
 
   const priceSymbol = useObservableState(selectedPricePoolAssetSymbol$, O.none)
 
   const priceRatio = useObservableState(priceRatio$, ONE_BN)
 
+  const isHistoryLoading = useObservableState(isHistoryLoading$, false)
+
   const poolDetailRD: PoolDetailRD = useObservableState(selectedPoolDetail$, RD.initial)
 
-  return FP.pipe(
-    RD.combine(routeAssetRD, poolDetailRD),
-    RD.fold(
-      renderInitialView,
-      renderPendingView,
-      (e: Error) => {
-        return <PoolStatus label={intl.formatMessage({ id: 'common.error' })} displayValue={e.message} />
-      },
-      ([asset, poolDetail]) => {
-        return (
-          <PoolDetails
-            asset={O.some(asset)}
-            depth={getDepth(poolDetail, priceRatio)}
-            volume24hr={get24hrVolume(poolDetail, priceRatio)}
-            allTimeVolume={getAllTimeVolume(poolDetail, priceRatio)}
-            totalSwaps={0 /* getTotalSwaps(history) */}
-            totalStakers={0 /* getTotalStakers(poolDetail) */}
-            priceUSD={getPriceUSD(poolDetail, priceRatio)}
-            priceSymbol={O.toUndefined(priceSymbol)}
-          />
+  const onRefreshData = useCallback(() => {
+    reloadSelectedPoolDetail()
+    reloadActionsHistory()
+  }, [reloadSelectedPoolDetail, reloadActionsHistory])
+
+  const refreshButtonDisabled = useMemo(() => {
+    return isHistoryLoading || FP.pipe(poolDetailRD, RD.isPending)
+  }, [isHistoryLoading, poolDetailRD])
+
+  const prev = useRef<React.ReactElement>()
```

Comment:
Not sure if this is really a good idea to store such a heavy component into a `Ref` ... What's the reason for this? Maybe another way could be storing all props of the component, but not the component itself. Also, `PoolDetailsView` has already a `loading` prop, why not just using it?